cli/src/utils/initHelloWorldApp.js

```diff
@@ -1,75 +1,82 @@
-import fs from 'node:fs';
+import fs from 'node:fs/promises';
 import path from 'node:path';
 import { fileURLToPath } from 'node:url';
-import util from 'util';
 import { copy } from './copy.js';
-
-const writeFileAsync = util.promisify(fs.writeFile);
+import { CONFIG_FILE, TEST_INPUT_DIR } from '../config/config.js';
 
 export async function initHelloWorldApp({
   projectName,
   hasProtectedData,
   template,
 }) {
   try {
+    // Copy template
     if (hasProtectedData) {
-      copyChosenTemplateFiles({
+      await copyChosenTemplateFiles({
         projectName,
         template: `withProtectedData/${template}`,
       });
     } else {
-      copyChosenTemplateFiles({
+      await copyChosenTemplateFiles({
         projectName,
         template: `withoutProtectedData/${template}`,
       });
     }
 
-    // Create idapp.config.json
+    // Create other files
     await createConfigurationFiles({ projectName, hasProtectedData });
+    await createInputFolder();
   } catch (err) {
     console.log('Error during project initialization:', err);
     throw err;
   }
 }
 
-async function createConfigurationFiles({ projectName, hasProtectedData }) {
-  // Create a simple iDapp configuration file
-  const configContent = `{
-  "projectName": "${projectName}",
-  "dockerhubUsername": "",
-  "withProtectedData": ${hasProtectedData}
+async function createInputFolder() {
+  await fs.mkdir(TEST_INPUT_DIR, { recursive: true });
 }
-`;
 
-  await writeFileAsync('./idapp.config.json', configContent, 'utf8');
+async function createConfigurationFiles({ projectName, hasProtectedData }) {
+  // Create a simple iDapp configuration file
+  const configContent = {
+    projectName: projectName,
+    dockerhubUsername: '',
+    withProtectedData: hasProtectedData,
+  };
+  await fs.writeFile(
+    CONFIG_FILE,
+    JSON.stringify(configContent, null, 2),
```

Comment:
Indeed much cleaner here to define a normal object and then use `JSON.stringify` 👍